### ISSUE TITLE
implement user-created Actions like Xfce's clipman plugin

### ISSUE THREAD

**Issue by jvonhoff** [popov895/Clipman#40]
I realize this is a long shot, but I was hoping this plugin was inspired by Xfce's clipman, so I thought I'd ask!

In Xfce's clipman, [there's a window](https://docs.xfce.org/panel-plugins/xfce4-clipman-plugin/start#action_settings) that allows you to add a regex entry along with a list of commands and actions.  When the copied text matches the regex, it will pop open a list of the commands you set up, and let you choose one to run.

For example, if you created a regex for `https?\:\/\/.*imdb.com/title/tt\(0-9{5-7}\)` and entered a command like 
`curl -H "Content-Type: application/json" -X POST ... http://192.168.1.10:8310/api/movie?apikey=YOURAPIKEYHERE`
then any time you happened to copy a movie link from IMDb, then it could automatically pop up a window to let you click that command (in this case curling the captured part of the regex over to, say, Radarr).  I realize that popping open a menu, particularly near the cursor, might not be possible, but just having the pertinent commands (and not nonsense commands) show up would be excellent.

In my Xfce days, I used it a lot:
- copy file names with commands to rotate or resize JPGs
- copy walls of text with a command to post to Pastebin (not Paste to Pastebin :rofl: )
- I even had one that would convert text to a QR code
and things like that.  It's a really great productivity tool (with a somewhat steep learning curve), so I would love to see something like that in Gnome.

Cheers!

**Comment by jvonhoff** [comment]
I just noticed your comment in #11 about wanting to remain lightweight.  I can certainly appreciate that, so I understand if that's not in your desired scope. :crossed_fingers:

**Comment by popov895** [comment]
Right, I have no plans to implement such a feature (at least in the near future).

> nonsense commands

Which commands and why?

**Comment by jvonhoff** [comment]
> Which commands and why?

Just saying that the regex would only show "resize image" commands when the file/url you have copied ends with PNG|JPG , etc. rather than showing you commands that don't apply to the link/text you have selected.

> no plans to implement such a feature

Understood.  Thanks for your consideration.

**Comment by popov895** [comment]
Well, let's keep it open. It's in principle a worthy idea.
